### DISCUSSION TITLE
What is considered a large size for pypi server index

### DISCUSSION THREAD

**Discussion by athornboroughp4** [pypiserver/pypiserver#601] [General]
Hello we are running a issue in our local pypi index. We have noticed that when we install certain packages from our local pypi server. There is a timeout that happens on the client side. Looking at the server it rails to 100%. Looking at the size of our our python package index. It has a total size of 22 GB (All packages combined) Most are 400KB worst case.  A lot of these packages have baked in dll or other assets files. We use that since python is our main tool house for alot of our applicational work.

Anyways I don't have a thumb in the wind if that is alot or not. In terms of python software it is huge. But in terms of storage etc. That seems small. We are using a bog standard tower to serve these running debian. Finally We keep our old packages in there in case there is some requirements conflict. Not sure if that is considered best practice or not, but would love to hear your thoughts. Thanks!

**Comment by dee-me-tree-or-love** [comment]
Hey @athornboroughp4, thank you for starting this discussion! I am very sorry for the radio-silence.  

I'm a little uncertain to tell if this would indeed be much. Generally I don't think it should be an impossible use case, but I'm also far from having benchmarked its performance yet nor have done much profiling 🫣 But I'd be very interested to dig further into it! 

Could you share some more details about your setup and if you have had any updates lately? 
1. For example, do you experience a consistent request speed or are there evident spikes? 
2. What would you say is the average time for your requests to complete? 
3. Could you share the startup files/configuration details that you use to host and run your `pypi-server` instance? 
4. Where are the clients downloading the packages located (e.g. in the same network as the server, very far geographic location, etc.)?

This would just help to get a better idea of the context. 

Also, here's some long-lasting documentation piece we have on ["Serving Thousands of Packages"](https://github.com/pypiserver/pypiserver#serving-thousands-of-packages), I'm not sure but maybe this could be of some help too.
Other than that I'd love to start doing some more profiling and benchmarking but it might unfortunately take a while to get there.   

I'd be very happy to hear more about your situation and try to gradually work towards more clarity on expected performance patterns. 

_P.S: Combining `pypi-server` with my other (work and research) makes my response rate sometimes a bit low, I'm sorry about this 😞 but I'm very eager to try to zoom into this._
